File: ucare_backend/src/main/java/com/douzone/ucare/service/DiseaseService.java
package com.douzone.ucare.service;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.douzone.ucare.repository.DiseaseRepository;
import com.douzone.ucare.vo.DiseaseVo;

@Service
public class DiseaseService {
	@Autowired
	private DiseaseRepository diseaseRepository;
	
	public int create(DiseaseVo data) {
		return diseaseRepository.create(data);
	}

	public List<DiseaseVo> retrieveAll() {
		return diseaseRepository.retrieveAll();
	}

	public int update(DiseaseVo data) {
		return diseaseRepository.update(data);
	}

	public int delete(int diseaseNo) {
		return diseaseRepository.delete(diseaseNo);
	}

}

File: ucare_frontend/src/service/diseaseService.js
import axios from 'axios';
import { DISEASE_API_BASE_URL } from './urlConfig'

class diseaseService {
  create(data){
    return axios.post(DISEASE_API_BASE_URL + '/create', data);
  }
  retrieveAll(){
    return axios.get(DISEASE_API_BASE_URL + '/retrieveAll');
  }
  update(data){
    return axios.put(DISEASE_API_BASE_URL + '/update', data)
  }
  delete(diseaseNo){
    return axios.delete(DISEASE_API_BASE_URL + '/delete/' + diseaseNo);
  }
}

export default new diseaseService();